src/main/java/gregtech/api/fluids/GTFluidRegistrator.java

```diff
@@ -0,0 +1,301 @@
+     *
+     * @param material        the material whose fluid to get
+     * @param type            the type for the fluid
+     * @param alternativeName the alternative name
+     */
+    public static void addAlternativeFluidName(@Nonnull Material material, @Nonnull FluidType type, @Nonnull String alternativeName) {
+        addAlternativeFluidName(type.getFluidNameForMaterial(material), alternativeName);
+    }
+
+    /**
+     * Add an alternative name for a fluid, to use if found instead of the default
+     *
+     * @param registeredName  the default name of the fluid
+     * @param alternativeName the alternative name
+     */
+    public static void addAlternativeFluidName(@Nonnull String registeredName, @Nonnull String alternativeName) {
+        Set<String> set = alternativeFluidNames.get(registeredName);
+        if (set == null) {
+            set = new ObjectOpenHashSet<>();
+            alternativeFluidNames.put(registeredName, set);
+        }
+        set.add(alternativeName);
+    }
+
+    /**
+     * Remove all alternative names for a fluid
+     *
+     * @param registeredName the fluid's registered name
+     */
+    @SuppressWarnings("unused")
+    public static void removeAlternativeFluidName(@Nonnull String registeredName) {
+        alternativeFluidNames.remove(registeredName);
+    }
+
+    /**
+     * Remove an alternative name for a fluid
+     *
+     * @param registeredName  the fluid's registered name
+     * @param alternativeName the name to remove
+     */
+    @SuppressWarnings("unused")
+    public static void removeAlternativeFluidName(@Nonnull String registeredName, @Nonnull String alternativeName) {
+        Set<String> set = alternativeFluidNames.get(registeredName);
+        if (set != null) set.remove(alternativeName);
+    }
+
+    /**
+     * Registers a fluid texture for use
+     *
+     * @param textureLocation the location of the texture to use
+     */
+    public static void registerFluidTexture(@Nonnull ResourceLocation textureLocation) {
+        fluidSprites.add(textureLocation);
+    }
+
+    /**
+     * @param fluidName the name of the fluid
+     * @return the fluid associated with the name
+     */
+    @Nullable
+    public static Fluid getExistingFluid(@Nonnull String fluidName) {
+        // if the fluid already exists
+        Fluid fluid = FluidRegistry.getFluid(fluidName);
+        if (fluid != null) return fluid;
+
+        // check if fluid is registered from elsewhere under an alternative name, if not already found
+        if (alternativeFluidNames.containsKey(fluidName)) {
+            for (String altName : alternativeFluidNames.get(fluidName)) {
+                // if the fluid already exists
+                fluid = FluidRegistry.getFluid(altName);
+                if (fluid != null) return fluid;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Create a standard Fluid compatible with the GT Fluid System from a definition.
```

Comment:
You should mentioned that it will not register fluids if one with same name already exists, but returns that one instead